DeusEx/Classes/DXRAugmentations.uc

```diff
@@ -57,3 +67,24 @@ function static Name PickRandomAug(DXRando dxr)
     log("Picked Aug "$ slot $"/"$numAugs$" " $ dxr.Player.AugmentationSystem.augClasses[slot].Name, 'DXRAugmentations');
     return dxr.Player.AugmentationSystem.augClasses[slot].Name;
 }
+
+function RandoAug(Augmentation a)
+{
+    local int i;
+    local float min;
+    local string s;
+
+    dxr.SetSeed( dxr.Crc(dxr.seed $ "RandoAug " $ a.class.name ) );
+
+    s = "(Values: ";
+    for(i=0; i<ArrayCount(a.LevelValues); i++) {
+        a.LevelValues[i] = a.default.LevelValues[i] * (rngf()+1.5)/2;
+        if( a.LevelValues[i] < min ) a.LevelValues[i] = min;
+        min = a.LevelValues[i];
+        if( i>0 ) s = s $ ", ";
+        s = s $ a.LevelValues[i];
+    }
+    s = s $ ")";
+    a.EnergyRate = a.default.EnergyRate * (rngf()+1.5)/2;
+    a.Description = a.Description $ "|n|n" $ s;
```

Comment:
this will probably stack the description? I think I could use the default.Description, but that might break alternate languages